docs/articles/nunit/release-notes/Nunit4.0-MigrationGuide.md

```diff
@@ -0,0 +1,169 @@
+Assert.That(actualText, Does.StartWith("42"), $"Expected '{actualText}' to start with '42'");
+```
+
+There are no code fixers for `FileAssert` and `DirectoryAssert`.
+They could be added, but we don't expect these to be used too much.
+
+### Convert Classic Assert into NUnit 4.x equivalent
+
+If you want to keep the Classic Asserts and not convert them to the constraint model --
+but do want to use the new NUnit 4.x naming -- you'll need to update the code manually.
+
+The NUnit.Analyzer can't help here as the code either doesn't compile before the change or after,
+depending on what version of `nunit` you are compiling with.
+
+If you _only_ use classic asserts, you can get away with a couple of global substitutes:
+
+1. Convert `using NUnit.Framework;` into _both_ `using NUnit.Framework;using NUnit.Framework.Legacy;`
+
+   Depending on your editor you can automatically insert a newline between the two `using` statements.
+1. Convert `Assert` into `ClassicAssert`.
+
+   This global substitute will also convert those asserts that have not changed.
+You can narrow the scope of this substitute to do only the asserts that need converting, but there are quite a lot.
+
+   1. Convert `Assert.AreEqual` into `ClassicAssert.AreEqual`.
+   1. Convert `Assert.True` into `ClassicAssert.True`.
+   1. Similar for `IsTrue`, `False`, `IsFalse`, `Greater`, `Less`, ...
+
+   Depending on what is less work, alternatively you can reverse the substitution of those that shouldn't be have been changed:
+   1. Convert `ClassicAssert.That` into `Assert.That`.
+   1. Convert `ClassicAssert.Fail` into `Assert.Fail`.
+   1. Convert `ClassicAssert.Throws` into `Assert.Throws`.
+   1. Etc.
+
+   Or if you use Visual Studio, it will raise an [IDE0002](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0002) with a code fix that can convert all of those that are not considered _classic_ back to assert in one swoop:
+
+   ![ClassicAssert to Assert](~/images/IDE0002-ClassicAssert-into-Assert.png)
+
+### Use `global using` aliases
+
+If you use SDK 6.0 or newer that supports C#10, you can upgrade without modifying any actual tests by adding the following aliases to `GlobalUsings.cs`
+
+```csharp
+global using Assert = NUnit.Framework.Legacy.ClassicAssert;
+global using CollectionAssert = NUnit.Framework.Legacy.CollectionAssert;
+global using StringAssert = NUnit.Framework.Legacy.StringAssert;
+global using DirectoryAssert = NUnit.Framework.Legacy.DirectoryAssert;
+global using FileAssert = NUnit.Framework.Legacy.FileAssert;
+```
+
+Note that this doesn't mean you have to target .NET 6.0. This also works if targeting .NET Framework as it is purely done on the source code level.
+
+## Assert.That with _format_ specification and `params` overload conversion
+
+These overloads were removed to allow for better messages in case of failure. See [The "Towards NUnit 4" article](../Towards-NUnit4.md#improved-assert-result-messages) for more information.
```

Comment:
You've made this one very long line.
Maybe a lint setting for a maximum line length would be good.